[IRTX] Use parse argument functions for IR TX command arguments

TD-er opened the pull request:
As mentioned here: https://github.com/letscontrolit/ESPEasy/issues/2724#issuecomment-735778417

TD-er committed:
[IRTX] Use parse argument functions for IR TX command arguments

As mentioned here: https://github.com/letscontrolit/ESPEasy/issues/2724#issuecomment-735778417

TD-er commented:
@jimmys01 

Can you try this test build?
- [ESPEasy_ESP32_mega-20201130-3-PR_3403.zip](https://www.dropbox.com/s/rgk4qda90x070xn/ESPEasy_ESP32_mega-20201130-3-PR_3403.zip?dl=0)
- [ESPEasy_ESP82xx_mega-20201130-3-PR_3403.zip](https://www.dropbox.com/s/dayp8wexthknyva/ESPEasy_ESP82xx_mega-20201130-3-PR_3403.zip?dl=0)

jimmys01 commented on src/_P035_IRTX.ino:
```diff
@@ -164,18 +164,18 @@ boolean Plugin_035(byte function, struct EventStruct *event, String &command)
 
   StaticJsonDocument<200> docTemp;
-  DeserializationError error = deserializeJson(docTemp, cmd.substring(cmd.indexOf(',') + 1, cmd.length()));
+  DeserializationError error = deserializeJson(docTemp, parseStringToEnd(cmd, 2));
```
cmd.length() is missing.

TD-er replied:
But that's only needed for substring, right?
parseStringToEnd is doing all the magic for you, including stripping matching quotes if present.
It does also convert to lower case.

jimmys01 commented:
> 
> 
> @jimmys01
> 
> Can you try this test build?
> 
>     * [ESPEasy_ESP32_mega-20201130-3-PR_3403.zip](https://www.dropbox.com/s/rgk4qda90x070xn/ESPEasy_ESP32_mega-20201130-3-PR_3403.zip?dl=0)
> 
>     * [ESPEasy_ESP82xx_mega-20201130-3-PR_3403.zip](https://www.dropbox.com/s/dayp8wexthknyva/ESPEasy_ESP82xx_mega-20201130-3-PR_3403.zip?dl=0)

give me 30 min

jimmys01 commented:
Same behavior. 
At least for the IRSENDAC command

jimmys01 commented:
```
#ifdef P016_P035_USE_RAW_RAW2
          handleRawRaw2Encoding(command);
#endif //P016_P035_USE_RAW_RAW2
          handleIRremote(command);
        }
```

Also lines 145 to 149 need your special treatment?
I am to sleep deprived right now, I will have a better look tomorow

TD-er committed:
[IRTX] Convert more commands to use parse string functions

TD-er commented:
Added the changes to those functions too.

TD-er committed:
Merge branch 'mega' into bugfix/Parse_IR_commands

jimmys01 commented:
Works, thank you.

Also the custom_ir build seems broken once more.

```
xtensa-lx106-elf-g++: error: CreateProcess: No such file or directory
*** [.pio\build\custom_IR_ESP8266_4M1M\ESP_Easy_mega_20201201_custom_IR_ESP8266_4M1M.elf] Error 1
```

TD-er commented:
Yep I noticed.
We're constantly hitting the max. command line limit in Windows.
I will poke the platformIO developers onece more to stress the importance of a work-around on that issue.

TD-er commented:
@jimmys01 I managed to make a nice work-around for the `longcmd` bug.
See: https://github.com/platformio/platform-espressif8266/issues/231

Just performing a few last tests and if those don't show serious issues, I will merge my PR for this so you can build it all again on Windows.

TD-er commented:
@jimmys01 I merged the PR so please let me know if you run into build issues again.

Does this PR need anything else or is it now working?

TD-er committed:
Merge branch 'mega' into bugfix/Parse_IR_commands

jimmys01 commented:
just to be sure the, command
IRSEND,RAW2,something,38,something
must be transformed to
IRSEND,'RAW2','something','38','something' ?

TD-er commented:
quotes are only needed if the parameter contains a character that can be considered a parameter separator, like:
- `,` (a comma)
- ` ` (a space)

And to make it possible to still use single or double quotes in a parameter, you can chose either of the other quote types to wrap a parameter.

So if you only have parameters that don't have a comma or a space, then you don't need to add quotes.

jimmys01 commented:
**We are good to go!** also the Custom IR builds work again!! Extremely good work, it seemed a really tricky bug to solve.
One minor issue is that if you press submit in the following screen


![εικόνα](https://user-images.githubusercontent.com/17748323/101021754-ffddb680-3578-11eb-9c3f-dca3c547555f.png)


the updated page will be missing the parameters and only hold the first part of the command.


![εικόνα](https://user-images.githubusercontent.com/17748323/101021876-31ef1880-3579-11eb-99a3-a01936a2d2bc.png)

just reporting it because i noticed it, nothing really bad comes out of it.

TD-er committed:
[Web UI] Proper encoding for HTML attribute values

All caused by this little remark: https://github.com/letscontrolit/ESPEasy/pull/3404#issuecomment-737972401

TD-er commented:
@jimmys01 
If you ever have "just reporting" feeling itching... Please be aware what it may result into ;)
See the last commit I added to this PR.

TD-er committed:
[Cleanup] Uncrustify of web page code
